apollo-router/src/services/router_service.rs

```diff
@@ -400,7 +400,10 @@ impl PluggableRouterServiceBuilder {
             .ok()
             .and_then(|x| x.parse().ok())
             .unwrap_or(100);
-        let query_cache = Arc::new(QueryCache::new(query_cache_limit, self.schema.clone()));
+        let query_cache = Arc::new(QueryCache::new(
+            query_cache_limit,
+            Arc::new((*self.schema.api_schema()).clone()),
```

Comment:
Does this indicate that it may be good to change struct Schema:
```
api_schema: Option<Arc<Schema>>,
```
instead of:
```
api_schema: Option<Box<Schema>>,
```
?

Reply:
right, good idea, I'll change that

Reply:
this is trickier than I thought, because api_schema either returns self, or the api_schema, and we don't have arbitrary self types. I'll do that in a follow up PR with another change I'd like to do: having the `Query` struct hold the schema (so we don't need to pass it explicitely to format_response)